Add io.github.denzilferreira.asteroid_shooter

denzilferreira opened the pull request:
<!-- ⚠️⚠️  Submission pull request MUST be made against the `new-pr` **base branch** ⚠️⚠️  -->

<!-- 💡 Please go to the preview tab to view the markdown below 💡 -->

### Please confirm your submission meets all the criteria

<!-- 💡 Please replace each `[ ]` with `[X]` when the step is complete 💡 -->

<!-- 💡 Please tick and write 'N/A' with a reason if a checklist item below is not applicable 💡 -->

- [x] Please describe the application briefly. 
An arcade, local 2-player game where you must survive for 120 seconds the asteroids.
- [x] Please attach a video showcasing the application on Linux using the Flatpak. [![YouTube](https://img.youtube.com/vi/qAgukJtbTuo/hqdefault.jpg)](https://youtu.be/qAgukJtbTuo)
- [x] The Flatpak ID follows all the rules listed in the [Application ID requirements][appid].
- [x] I have read and followed all the [Submission requirements][reqs] and the [Submission guide][reqs2] and I agree to them.
- [x] I am the author/developer contributor to the project.

<!-- 💡 Please mention below the GitHub usernames of any additional maintainers needed (if any) 💡 -->

<!-- ⚠️⚠️  Please DO NOT modify anything below this line ⚠️⚠️  -->

[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[reqs2]: https://docs.flathub.org/docs/for-app-authors/submission

hfiguiere commented on io.github.denzilferreira.asteroid_shooter.desktop:
this file should be (submitted and managed) upstream

hfiguiere commented on io.github.denzilferreira.asteroid_shooter.metainfo.xml:
this file should be (submitted and managed) upstream

hfiguiere commented on io.github.denzilferreira.asteroid_shooter.png:
this file should be (submitted and managed) upstream

hfiguiere commented on io.github.denzilferreira.asteroid_shooter.json:
```diff
@@ -0,0 +1,61 @@
+      ],
+      "sources": [
+        {
+          "type": "dir",
```
never use source `type: dir`

denzilferreira replied:
first time submitting to FlatHub :blush: The source is on my repo, so now I see this needs to be type git :+1: I'll update this.

denzilferreira commented:
bot, build

denzilferreira commented:
bot, build

denzilferreira commented on io.github.denzilferreira.asteroid_shooter.json:
```diff
@@ -0,0 +1,44 @@
+        {
+          "type": "git",
+          "url": "https://github.com/denzilferreira/asteroid_shooter.git",
+          "tag": "v1.0.0",
```
Yep, just realised I forgot to update the release. Updated now, should pass.

denzilferreira commented:
bot, build

hfiguiere commented on io.github.denzilferreira.asteroid_shooter.json:
```diff
@@ -0,0 +1,46 @@
+    {
+      "name": "asteroid-shooter",
+      "buildsystem": "simple",
+      "build-commands": [],
```
not sure how it can build fine locally.

hfiguiere replied:
again I don't see how this build fine. Maybe you are confusing what "build it locally" means.

hfiguiere commented:
there is no build commands. so nothing is done.

hfiguiere commented:
`buildsystem: simple` need build commands.

hfiguiere commented on io.github.denzilferreira.asteroid_shooter.json:
```diff
@@ -39,7 +39,7 @@
           "url": "https://github.com/denzilferreira/asteroid_shooter.git",
           "tag": "v1.0.0",
-          "commit": "c1724ba407d09006d2598809116970252bf1c95b"
+          "commit": "39d37ecad77812a12a68ee552fa13beeb86dec7a"
```
retagging is bad.

denzilferreira replied:
Should I make a new release with a new tag?... v1.0.0 is not really released yet...

denzilferreira replied:
fair enough.

denzilferreira commented:
> "metainfo-launchable-tag-wrong-value: Launchable tag in Metainfo is wrong: io.github.denzilferreira.android_shooter.desktop"

so why does the documentation have .desktop file as an example? https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines#launchable it explicitly says it needs to refer to a .desktop file :confused:

denzilferreira commented:
bot, build

hfiguiere commented:
> > "metainfo-launchable-tag-wrong-value: Launchable tag in Metainfo is wrong: io.github.denzilferreira.android_shooter.desktop"
> 

Really look closely at the word before `_shooter`. Really. Put the app-id next to it. And again. Until you see something. It's not even the same number of letters. But it rhymes.

hfiguiere commented on io.github.denzilferreira.asteroid_shooter.json:
```diff
@@ -32,13 +32,13 @@
       "buildsystem": "simple",
       "build-commands": [
-        "bash ./build.sh"
+        "bash ./build.sh --prefix=${FLATPAK_DEST}"
```
I have no idea what you are trying to do here.

denzilferreira replied:
was testing to see if this was something to do with where the flatpak was being installed on the CICD and not being able to find the .desktop file (the android vs asteroid :see_no_evil: )

denzilferreira commented:
bot, build

denzilferreira commented on io.github.denzilferreira.asteroid_shooter.json:
```diff
@@ -32,13 +32,13 @@
           "type": "git",
           "url": "https://github.com/denzilferreira/asteroid_shooter.git",
-          "branch": "main"
+          "tag": "v1.0.0"
```
removed the tag, kept commit

denzilferreira commented:
> ✅ [Test build succeeded](https://github.com/flathub-infra/vorarbeiter/actions/runs/15911168515). To test this build, install it from the testing repository:
> 
> ```
> flatpak install --user https://dl.flathub.org/build-repo/194181/io.github.denzilferreira.asteroid_shooter.flatpakref
> ```

it works!!! :tada:

bbhtt commented:
you took the easy way out and hardcoded "/app" in various places https://github.com/denzilferreira/asteroid_shooter/blob/f88ec2bd8316d5b6accca693d6d8e40994b095de/spaceshooter.py#L365-L372

so i guess this is unsable elsewhere.

bbhtt commented:
why does the requirements.txt get installed to `/app` https://github.com/denzilferreira/asteroid_shooter/blob/f88ec2bd8316d5b6accca693d6d8e40994b095de/build.sh#L11 ?

denzilferreira commented on io.github.denzilferreira.asteroid_shooter.json:
```diff
@@ -0,0 +1,46 @@
+  "sdk": "org.freedesktop.Sdk",
+  "command": "asteroid-shooter",
+  "finish-args": [
+    "--share=network",
```
I'll test without. I believe this was for gamepads connected with Bluetooth, but I'm not entirely sure. I'll confirm.

denzilferreira commented on io.github.denzilferreira.asteroid_shooter.json:
```diff
@@ -0,0 +1,46 @@
+    "--share=network",
+    "--socket=fallback-x11",
+    "--socket=wayland",
+    "--device=dri",
+    "--device=all",
```
Actually only need input for wired gamepads and dri for graphics. Will update.

bbhtt commented:
> Ah 😯 good catch. I'll replace that with flatpak destination environment variable

No the point is this should not be specific to Flatpak. What if someone tries to install outside Flatpak?

The `/app` path does not exist

denzilferreira commented:
> > Ah 😯 good catch. I'll replace that with flatpak destination environment variable
> 
> No the point is this should not be specific to Flatpak. What if someone tries to install outside Flatpak?
> 
> The `/app` path does not exist

Right, right. I should use the current folder where the script is located, updated and tested locally and works. Built the flatpak locally and it also works. Thanks!

denzilferreira commented:
bot, build

denzilferreira commented:
> ✅ [Test build succeeded](https://github.com/flathub-infra/vorarbeiter/actions/runs/15930442301). To test this build, install it from the testing repository:
> 
> ```
> flatpak install --user https://dl.flathub.org/build-repo/194330/io.github.denzilferreira.asteroid_shooter.flatpakref
> ```

Just tested, it is working.

denzilferreira commented:
bot, build

bbhtt commented:
Also, it is better to install everything to a subdirectory of app, `/app/spaceshooter`.

And I didn't understand why a second script is needed to launch. The game is a single file script, why not make it launchable directly?

denzilferreira commented:
> It's still there https://github.com/denzilferreira/asteroid_shooter/blob/f88ec2bd8316d5b6accca693d6d8e40994b095de/build.sh#L11

it's not on the built commit: https://github.com/denzilferreira/asteroid_shooter/blob/01271f24cb2a7fc4021a2213fe30d53feb3ccaea/build.sh

denzilferreira commented:
> Also, it is better to install everything to a subdirectory of app, `/app/spaceshooter`.
> 
> And I didn't understand why a second script is needed to launch. The game is a single file script, why not make it launchable directly?

I'm not sure how to accomplish this using a Python app. Changing chmod to +x and running it (./spaceshooter.py) does not work :eyes:  We still need to have the /app/bin exec right?

denzilferreira commented:
@bbhtt I added the shebang to the spaceshooter.py after moving all the source to /app/spaceshooter folder. This causes an issue on the location of the /app/bin executable and the source. The source is in /app/spaceshooter/spaceshooter.py but when calling the /app/bin/asteroid-shooter the parent folder is not the same as spaceshooter.py. The only way to get this to work both by calling the script directly and via the /app/bin executable was to use a bash script that then calls the executable on the right folder on Flatpak. I'm not sure if there is a better way to handle this. I'm open to suggestions.

I've updated the screenshots to also now include the window borders (Gnome's default screenshot tool does not render them, unless I use Alt+PrintScreen, which is what I used now).

bbhtt commented:
You don't have a license file here https://github.com/denzilferreira/asteroid_shooter but metainfo says Apache.

denzilferreira commented:
bot, build

denzilferreira commented:
> You don't have a licence file here https://github.com/denzilferreira/asteroid_shooter but metainfo says Apache-2.0.

Added :+1:
